src/cljs/org/broadinstitute/firecloud_ui/page/workspace/summary/library.cljs

```diff
@@ -58,6 +61,33 @@
     (do (utils/log "unknown type: " type)
         value)))
 
+(defn select2-typeahead [placeholder r on-item-selected]
+  (.select2 (js/$ r)
+            (clj->js { :ajax { :url (fn [params]
+                                      (str (config/api-url-root) "/duos-autocomplete/" (aget params "term")))
+                               :dataType "json"
+                               :delay 250
+                               :processResults (fn [data]
+                                                 (clj->js {:results data}))
+                               :data (fn [params] ;; don't actually want to send it as a query
+                                       (clj->js {}))}
+                       :minimumInputLength 1
+                       :cache true
+                       :templateResult (fn [obj]
+                                         (if (aget obj "loading")
+                                           (aget obj "text")
+                                           (str "<div style='font-weight: bold'>" (aget obj "label")
+                                                "<div style='float: right; font-weight: normal'> DOID: "
+                                                (style/url-to-doid (aget obj "id")) "</div></div>"
+                                                "<div style='font-size: small'> "
+                                                (if (not (nil? (aget obj "definition")))
+                                                  (aget obj "definition") "</div>"))))
```

Comment:
Uhh I don't think that's possible... The select2 doesn't know how to go from cljs to html and I know this is kinda gross but I'm not sure how to change it..